service_container/expression_language.rst

```diff
@@ -125,3 +125,54 @@ via a ``container`` variable. Here's another example:
 
 Expressions can be used in ``arguments``, ``properties``, as arguments with
 ``configurator`` and as arguments to ``calls`` (method calls).
+
+You can also use expressions as service factories:
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # config/services.yaml
+        services:
+            App\Mailer:
+                factory: "@=parameter('some_param') ? service('some_service') : arg(0)"
```

Comment:
Does this require the class referenced by the `some_service` to have a `__invoke()` method?

Reply:
No, it doesn't use the result of the expression as of factory but the expression itself is the factory. In this case this means hat the `App\Mailer` service will be an alias to `some_service` if the parameter `some_param` is `true`.